src/crypto/crypto_dh.cc

```diff
@@ -552,16 +551,12 @@ EVPKeyCtxPointer DhKeyGenTraits::Setup(DhKeyPairGenConfig* params) {
     if (!dh)
       return EVPKeyCtxPointer();
 
-    BIGNUM* prime = prime_fixed_value->get();
-    BignumPointer bn_g(BN_new());
-    if (!BN_set_word(bn_g.get(), params->params.generator) ||
-        !DH_set0_pqg(dh.get(), prime, nullptr, bn_g.get())) {
+    auto bn_g = BignumPointer::Create();
+    if (!bn_g.setWord(params->params.generator) ||
```

Comment:
ditto. this if statement can be merged with previous line.

Reply:
This one I'd rather keep separate given the inversion of the check (`if (!bn_g....`) This is a bit more readable for me.